tests/python/unittest/test_custom_datatypes_change_dtype.py

```diff
@@ -0,0 +1,481 @@
+    register_op(create_lower_func({32: 'Nop32Exp'}),
+                "Call",
+                "llvm",
+                "noptype",
+                intrinsic_name="exp")
+    register_op(create_lower_func({32: 'Nop32Log'}),
+                "Call",
+                "llvm",
+                "noptype",
+                intrinsic_name="log")
+    register_op(create_lower_func({32: 'Nop32Sigmoid'}),
+                "Call",
+                "llvm",
+                "noptype",
+                intrinsic_name="sigmoid")
+    register_op(create_lower_func({32: 'Nop32Tanh'}),
+                "Call",
+                "llvm",
+                "noptype",
+                intrinsic_name="tanh")
+    # This can be anything, considering the type isn't functionally correct.
+    register_min_func(lambda num_bits: 0, "noptype")
+
+
+def run_ops(src_dtype, dst_dtype, rtol=1e-7, atol=1e-7):
+    """Run the same op, but with two different datatypes"""
+    def check_unary_op(op, src_dtype, dst_dtype):
+        t1 = relay.TensorType((5, 10, 5), src_dtype)
+        x = relay.var("x", t1)
+        z = op(x)
+        x_data = rs.rand(5, 10, 5).astype(t1.dtype)
+
+        module = tvm.IRModule.from_expr(relay.Function([x], z))
+
+        compare(module, (x_data, ), src_dtype, dst_dtype, rtol, atol)
+        # print(maybe_correct_converted)
+        # print(correct)
+
+    for op in [
+            relay.nn.softmax,
+            tvm.relay.log,
+            tvm.relay.exp,
+            tvm.relay.sqrt,
+            tvm.relay.rsqrt,
+            tvm.relay.sigmoid,
+            tvm.relay.tanh,
+            relay.nn.relu,
+    ]:
+        check_unary_op(op, src_dtype, dst_dtype)
+
+    def check_binary_op(opfunc, src_dtype, dst_dtype):
+        t1 = relay.TensorType((5, 10, 5), src_dtype)
+        t2 = relay.TensorType((5, ), src_dtype)
+        x = relay.var("x", t1)
+        y = relay.var("y", t2)
+        z = opfunc(x, y)
+        x_data = rs.rand(5, 10, 5).astype(t1.dtype)
+        y_data = rs.rand(5).astype(t2.dtype)
+        module = tvm.IRModule.from_expr(relay.Function([x, y], z))
+
+        compare(module, (x_data, y_data), src_dtype, dst_dtype, rtol, atol)
+
+    for op in [
+            relay.add,
+            relay.subtract,
+            relay.divide,
+            relay.multiply,
+    ]:
+        check_binary_op(op, src_dtype, dst_dtype)
+
+    # we would like to test tvm_if_then_else
+    # but Relay.IfNode is not lowered to this intrinsic,
+    # so to keep our tests consistent with relay, we decide to not unit test
+    # Note: tvm_if_then_else is tested as part of the mobile_net model
+
+
+def run_model(get_workload,
+              input_shape,
+              src_dtype,
+              dst_dtype,
+              num_classes,
+              rtol=0.0001,
+              atol=0.0001):
+    module, params = get_workload(image_shape=input_shape,
+                                  num_classes=num_classes)
+
+    # Convert the input into the correct format.
```

Comment:
"Convert" might not be the right word, as we're not actually converting existing data, but generating random data.

Reply:
changed to "generate random input with appropriate shape/type"